staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -225,6 +220,11 @@ func (g *Cloud) ensureInternalLoadBalancer(clusterName, clusterID string, svc *v
 		}
 	}
 
+	// Ensure firewall rules if necessary
+	if err = g.ensureInternalFirewalls(loadBalancerName, ipToUse, clusterID, nm, svc, strconv.Itoa(int(hcPort)), sharedHealthCheck, nodes); err != nil {
```

Comment:
I don't think that it is necessary to query for Forwarding rule inside `ensureInternalFirewalls`. 
Maybe you can just assign ip address from forwarding rule to `ipToUse` if needed and remove the check for empty ipAddress inside the function.

Reply:
Updated.